inventory_management_system_api/services/item.py

```diff
@@ -118,9 +121,69 @@ def get(self, item_id: str) -> Optional[ItemOut]:
         :param item_id: The ID of the item to retrieve
         :return: The retrieved item, or `None` if not found
         """
-
         return self._item_repository.get(item_id)
 
+    def update(self, item_id: str, item: ItemPatchRequestSchema) -> ItemOut:
+        """
+        Update an item by its ID.
+
+        The method checks if the item exists in the database and raises a `MissingRecordError` if it does
+        not. If the catalogue item ID or system ID is being updated, it checks if a catalogue item
+        and/or system ID with such ID exists and raises a `MissingRecordError` if it does not.
```

Comment:
If I understand correctly I think this could do with a message along the lines of
`When updating properties, all existing properties that are different to those of the catalogue item must be provided or they will be overwritten by those in the catalogue item.`

I believe Joshua's edit PR supplies them in one go so this should currently be fine, although I would have initially expected it to just modify those given and leave the rest so that there is no overriding happening here. I would suggest either an explanation like this, or modify it to only replace those given ignoring the contents of the catalogue item. @VKTB any thoughts?


We also need a check like https://github.com/ral-facilities/inventory-management-system/blob/daf48454f38b202392f102aa68e7de8264412061/src/catalogue/items/catalogueItemDirectoryDialog.component.tsx#L83-L86 to prevent moving to a catalogue item with different properties but that is also part of #128 anyway.

Reply:
> If I understand correctly I think this could do with a message along the lines of `When updating properties, all existing properties that are different to those of the catalogue item must be provided or they will be overwritten by those in the catalogue item.`

I agree. The `properties` field should also be redeclared in the `ItemPatchRequestSchema` schema model class so that its description can mention this.

> I believe Joshua's edit PR supplies them in one go so this should currently be fine, although I would have initially expected it to just modify those given and leave the rest so that there is no overriding happening here. I would suggest either an explanation like this, or modify it to only replace those given, ignoring the contents of the catalogue item. @VKTB any thoughts?

I don't think the `PATCH` logic of any of the other endpoints update only those provided(?) so I would rather not do something different here as it would be confusing.